### PR TITLE
Mappings and Explanatory Javadoc for Chunk Update Changes

### PR DESCRIPTION
**Pull request by shartte**
There were some changes to chunk update since 1.16 and this PR tries to:

- Add some of the mappings related to it
- Add explanatory Javadoc where possible to make it easier for other people poking through that code

### CODE REVIEW

File: mappings/net/minecraft/network/packet/s2c/play/ChunkDataS2CPacket.mapping [hunk 1]
```diff
@@ -7,9 +7,16 @@ CLASS net/minecraft/class_2672 net/minecraft/network/packet/s2c/play/ChunkDataS2
 	FIELD field_12238 blockEntities Ljava/util/List;
 	FIELD field_16416 heightmaps Lnet/minecraft/class_2487;
 	FIELD field_20664 biomeArray [I
+	FIELD field_25720 skipLightingUpdate Z
```
Comment (liach) [inline mappings/net/minecraft/network/packet/s2c/play/ChunkDataS2CPacket.mapping]: `retainOldLighting`?

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping [hunk 2]
```diff
@@ -73,6 +75,34 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
 	METHOD method_22446 stream (Lnet/minecraft/class_1923;I)Ljava/util/stream/Stream;
 		ARG 0 center
 		ARG 1 radius
+	METHOD method_30551 getPackedLocalPosX (S)I
```
Comment (liach) [inline mappings/net/minecraft/util/math/ChunkSectionPos.mapping]: `unpack(To)LocalX`, so on

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping [hunk 3]
```diff
@@ -73,6 +75,34 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
...
+		ARG 0 packedLocalPos
+	METHOD method_30552 getPackedLocalPosY (S)I
+		COMMENT Gets the local y-coordinate from the given packed local position.
+		COMMENT @see #getPackedLocalPos
+		ARG 0 packedLocalPos
+	METHOD method_30553 getPackedLocalPosZ (S)I
+		COMMENT Gets the local z-coordinate from the given packed local position.
+		COMMENT @see #getPackedLocalPos
+		ARG 0 packedLocalPos
+	METHOD method_30554 getPackedLocalPosBlockX (S)I
```
Comment (liach) [inline mappings/net/minecraft/util/math/ChunkSectionPos.mapping]: `unpack(To)WorldX`, so on

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping (hunk 2 above)
Comment (shartte) [reply mappings/net/minecraft/util/math/ChunkSectionPos.mapping]: That would be inconsistent with the other methods, although I also like that more :-|
Also, "getPacked..." should be "packLocalPos", would you also favor that change together with renaming all of these to "unpack..."?

Comment (shartte) [reply mappings/net/minecraft/util/math/ChunkSectionPos.mapping]: Oh and also inconsistent with the functions to pack/unpack the position of chunk positions themselves.

Comment (liach) [reply mappings/net/minecraft/util/math/ChunkSectionPos.mapping]: Yeah `getPacked` is accessing a packed value. However there is no packed value to access here: the packed value is rather passed by the method caller and the unpacked value is returned. Hence `unpack` than `getPacked`, and other `getPacked` methods need fixing as well If they are unpacking a value than accessing a packed value.

Comment (shartte) [reply mappings/net/minecraft/util/math/ChunkSectionPos.mapping]: No, what I mean the other method is actually doing the packing :-D
But I take that as a yes.

Comment (shartte) [reply mappings/net/minecraft/util/math/ChunkSectionPos.mapping]: On second thought, I looked through the methods again and I'd rather leave them as is. The class has a shitton of methods dealing with various different coordinate systems, and I think being verbose here - while tedious - avoids mistakes.

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping (hunk 3 above)
Comment (shartte) [reply mappings/net/minecraft/util/math/ChunkSectionPos.mapping]: See my comment above. I'd rather keep these verbose names.

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/util/math/ChunkSectionPos.mapping]: So this method actually does packing? Are you kidding me, packing a short into an integer?

**Review by liach — CHANGES_REQUESTED** [review]
Unfortunately I refuse to approve this pr until we can sort out this pack and unpack mess. Current names make me feel like the returned values of those methods are packed, while it most likely is not packed since you don't pack a short into an integer.

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping (hunk 2 above)
Comment (shartte) [reply mappings/net/minecraft/util/math/ChunkSectionPos.mapping]: No, but the alternative is: `getXFromPackedLocalPos`.
Regarding pack/unpack: It's not consistent with any of the other methods in this class that do packing.
I.e. `asLong`...

Comment (shartte) [reply mappings/net/minecraft/util/math/ChunkSectionPos.mapping]: Okay, I refactored more members in this class and added alot more docs. Is this better?

**Review by liach — APPROVED** [review]
This looks so much better

File: mappings/net/minecraft/network/packet/s2c/play/ChunkDataS2CPacket.mapping [hunk 4]
```diff
@@ -25,3 +32,6 @@ CLASS net/minecraft/class_2672 net/minecraft/network/packet/s2c/play/ChunkDataS2
 	METHOD method_11530 isFullChunk ()Z
 	METHOD method_16123 getHeightmaps ()Lnet/minecraft/class_2487;
 	METHOD method_22422 getBiomeArray ()[I
+	METHOD method_30144 isSkipLightingUpdate ()Z
```
Comment (YanisBft) [inline mappings/net/minecraft/network/packet/s2c/play/ChunkDataS2CPacket.mapping]:
```suggestion
	METHOD method_30144 shouldSkipLightingUpdate ()Z
```

File: mappings/net/minecraft/network/packet/s2c/play/ChunkDeltaUpdateS2CPacket.mapping [hunk 5]
```diff
@@ -1 +1,16 @@
 CLASS net/minecraft/class_2637 net/minecraft/network/packet/s2c/play/ChunkDeltaUpdateS2CPacket
+	FIELD field_26345 sectionPos Lnet/minecraft/class_4076;
+	FIELD field_26346 packedLocalPos [S
+		COMMENT The packed local positions {@see ChunkSectionPos#getPackedLocalPos} for
+		COMMENT each entry in {@see #blockState}.
+	FIELD field_26347 blockState [Lnet/minecraft/class_2680;
+	METHOD <init> (Lnet/minecraft/class_4076;Lit/unimi/dsi/fastutil/shorts/ShortSet;Lnet/minecraft/class_2826;)V
+		ARG 1 sectionPos
+			COMMENT The position of the given chunk section that will be sent to the client.
```
Comment (YanisBft) [inline mappings/net/minecraft/network/packet/s2c/play/ChunkDeltaUpdateS2CPacket.mapping]:
```suggestion
			COMMENT the position of the given chunk section that will be sent to the client
```

File: mappings/net/minecraft/network/packet/s2c/play/ChunkDeltaUpdateS2CPacket.mapping [hunk 6]
```diff
@@ -1 +1,16 @@
...
+		ARG 2 updatedLocalPosSet
+			COMMENT The set of packed local positions within the given chunk section that should be included in the packet.
```
Comment (YanisBft) [inline mappings/net/minecraft/network/packet/s2c/play/ChunkDeltaUpdateS2CPacket.mapping]:
```suggestion
			COMMENT the set of packed local positions within the given chunk section that should be included in the packet
```

File: mappings/net/minecraft/server/world/ChunkHolder.mapping [hunk 7]
```diff
@@ -20,6 +20,13 @@ CLASS net/minecraft/class_3193 net/minecraft/server/world/ChunkHolder
 	FIELD field_17210 playersWatchingChunkProvider Lnet/minecraft/class_3193$class_3897;
 	FIELD field_19238 ticking Z
 	FIELD field_19333 tickingFuture Ljava/util/concurrent/CompletableFuture;
+	FIELD field_25803 pendingBlockUpdates Z
+		COMMENT Indicates that {@link #blockUpdatesBySection} contains at least one entry.
+	FIELD field_25804 blockUpdatesBySection [Lit/unimi/dsi/fastutil/shorts/ShortSet;
+		COMMENT Contains the packed chunk-local positions that have been marked for update
+		COMMENT by {@link #markForBlockUpdate}, grouped by their chunk y-section.
+		COMMENT <p>
+		COMMENT Entries for a section are null, if the section has no positions marked for update.
```
Comment (YanisBft) [inline mappings/net/minecraft/server/world/ChunkHolder.mapping]:
```suggestion
		COMMENT Entries for a section are null if the section has no positions marked for update.
```

File: mappings/net/minecraft/server/world/ChunkHolder.mapping [hunk 8]
```diff
@@ -50,6 +57,7 @@ CLASS net/minecraft/class_3193 net/minecraft/server/world/ChunkHolder
 	METHOD method_14012 markForLightUpdate (Lnet/minecraft/class_1944;I)V
 		ARG 1 type
 		ARG 2 y
+			COMMENT Chunk section y coordinate
```
Comment (YanisBft) [inline mappings/net/minecraft/server/world/ChunkHolder.mapping]:
```suggestion
			COMMENT chunk section y coordinate
```

File: mappings/net/minecraft/server/world/ChunkHolder.mapping [hunk 9]
```diff
@@ -63,14 +71,18 @@ CLASS net/minecraft/class_3193 net/minecraft/server/world/ChunkHolder
...
 	METHOD method_21737 getNowFuture (Lnet/minecraft/class_2806;)Ljava/util/concurrent/CompletableFuture;
 		ARG 1 leastStatus
+	METHOD method_23270 getCurrentStatus ()Lnet/minecraft/class_2806;
 	METHOD method_23271 getLevelType ()Lnet/minecraft/class_3193$class_3194;
+	METHOD method_30311 tryUpdateBlockEntityAt (Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;)V
 	CLASS class_3194 LevelType
 		METHOD method_14014 isAfter (Lnet/minecraft/class_3193$class_3194;)Z
 			ARG 1 levelType
 	CLASS class_3724 Unloaded
+		COMMENT This is used to represent a chunk that has not been loaded yet.
```
Comment (YanisBft) [inline mappings/net/minecraft/server/world/ChunkHolder.mapping]:
```suggestion
		COMMENT Used to represent a chunk that has not been loaded yet.
```

Comment (liach) [reply mappings/net/minecraft/server/world/ChunkHolder.mapping]:
```suggestion
		COMMENT Represents a chunk that has not been loaded yet.
```

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping [hunk 10]
```diff
@@ -5,51 +5,77 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
...
 		ARG 0 entity
 	METHOD method_18681 from (Lnet/minecraft/class_1923;I)Lnet/minecraft/class_4076;
+		COMMENT Creates a chunk section position from a chunk position and the y-coordinate of the vertical section.
 		ARG 0 chunkPos
 		ARG 1 y
 	METHOD method_18682 from (Lnet/minecraft/class_2338;)Lnet/minecraft/class_4076;
 		ARG 0 pos
 	METHOD method_18683 getSectionY ()I
 	METHOD method_18684 getLocalCoord (I)I
+		COMMENT Converts a world-coordinate to the local coordinate system (0-15) of its corresponding chunk section.
```
Comment (haykam821) [inline mappings/net/minecraft/util/math/ChunkSectionPos.mapping]:
```suggestion
		COMMENT Converts a world coordinate to the local coordinate system (0-15) of its corresponding chunk section.
```

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping [hunk 11]
```diff
@@ -5,51 +5,77 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
 		ARG 3 z
 	METHOD method_18674 getSectionX ()I
 	METHOD method_18675 getSectionCoord (I)I
+		COMMENT Converts a world-coordinate to the corresponding chunk-section coordinate.
```
Comment (haykam821) [inline mappings/net/minecraft/util/math/ChunkSectionPos.mapping]:
```suggestion
		COMMENT Converts a world coordinate to the corresponding chunk section coordinate.
```

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping [hunk 12]
```diff
@@ -5,51 +5,77 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
...
 		ARG 0 coord
 	METHOD method_18676 from (III)Lnet/minecraft/class_4076;
+		COMMENT Creates a chunk section position from its x, y and z coordinates.
```
Comment (haykam821) [inline mappings/net/minecraft/util/math/ChunkSectionPos.mapping]:
```suggestion
		COMMENT Creates a chunk section position from its X, Y, and Z coordinates.
```

File: mappings/net/minecraft/server/world/ChunkHolder.mapping [hunk 13]
```diff
@@ -20,6 +20,13 @@ CLASS net/minecraft/class_3193 net/minecraft/server/world/ChunkHolder
 	FIELD field_17210 playersWatchingChunkProvider Lnet/minecraft/class_3193$class_3897;
 	FIELD field_19238 ticking Z
 	FIELD field_19333 tickingFuture Ljava/util/concurrent/CompletableFuture;
+	FIELD field_25803 pendingBlockUpdates Z
+		COMMENT Indicates that {@link #blockUpdatesBySection} contains at least one entry.
+	FIELD field_25804 blockUpdatesBySection [Lit/unimi/dsi/fastutil/shorts/ShortSet;
+		COMMENT Contains the packed chunk-local positions that have been marked for update
+		COMMENT by {@link #markForBlockUpdate}, grouped by their chunk y-section.
```
Comment (haykam821) [inline mappings/net/minecraft/server/world/ChunkHolder.mapping]:
```suggestion
		COMMENT by {@link #markForBlockUpdate}, grouped by their chunk Y section.
```

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping [hunk 14]
```diff
@@ -5,51 +5,77 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
...
 	METHOD method_18684 getLocalCoord (I)I
+		COMMENT Converts a world-coordinate to the local coordinate system (0-15) of its corresponding chunk section.
 		ARG 0 coord
 	METHOD method_18685 asLong (III)J
 		ARG 0 x
 		ARG 1 y
 		ARG 2 z
-	METHOD method_18686 getX (J)I
+	METHOD method_18686 unpackX (J)I
+		COMMENT Gets the chunk section x-coordinate from the given packed chunk section coordinate.
```
Comment (haykam821) [inline mappings/net/minecraft/util/math/ChunkSectionPos.mapping:48]:
```suggestion
		COMMENT Gets the chunk section X coordinate from the given packed chunk section coordinate.
```

File: mappings/net/minecraft/server/world/ChunkHolder.mapping (hunk 8 above)
Comment (haykam821) [inline mappings/net/minecraft/server/world/ChunkHolder.mapping]:
```suggestion
			COMMENT Chunk section Y coordinate
```

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping [hunk 15]
```diff
@@ -5,51 +5,77 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
...
+		COMMENT @see #asLong
 		ARG 0 packed
 	METHOD method_18687 getSectionZ ()I
 	METHOD method_18688 getBlockCoord (I)I
+		COMMENT Converts the given chunk section coordinate to the world coordinate system.
+		COMMENT The returned coordinate will always be at the origin of the chunk section in world space.
 		ARG 0 sectionCoord
-	METHOD method_18689 getY (J)I
+	METHOD method_18689 unpackY (J)I
+		COMMENT Gets the chunk section y-coordinate from the given packed chunk section coordinate.
```
Comment (haykam821) [inline mappings/net/minecraft/util/math/ChunkSectionPos.mapping:57]:
```suggestion
		COMMENT Gets the chunk section Y coordinate from the given packed chunk section coordinate.
```

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping [hunk 16]
```diff
@@ -5,51 +5,77 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
...
+		COMMENT @see #asLong
 		ARG 0 packed
-	METHOD method_18690 getZ (J)I
+	METHOD method_18690 unpackZ (J)I
+		COMMENT Gets the chunk section z-coordinate from the given packed chunk section coordinate.
```
Comment (haykam821) [inline mappings/net/minecraft/util/math/ChunkSectionPos.mapping:61]:
```suggestion
		COMMENT Gets the chunk section Z coordinate from the given packed chunk section coordinate.
```

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping [hunk 17]
```diff
@@ -5,51 +5,77 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
...
 		ARG 0 packed
 	METHOD method_18691 fromBlockPos (J)J
+		COMMENT Gets the packed chunk section coordinate for a given packed {@link BlockPos}.
+		COMMENT @see #asLong
+		COMMENT @see BlockPos#asLong
 		ARG 0 blockPos
 	METHOD method_18692 toChunkPos ()Lnet/minecraft/class_1923;
-	METHOD method_18693 withZeroZ (J)J
+	METHOD method_18693 withZeroY (J)J
+		COMMENT Gets the packed chunk section coordinate at y=0 for the same chunk as
```
Comment (haykam821) [inline mappings/net/minecraft/util/math/ChunkSectionPos.mapping:71]:
```suggestion
		COMMENT Gets the packed chunk section coordinate at Y=0 for the same chunk as
```

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping [hunk 18]
```diff
@@ -73,6 +99,34 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
 	METHOD method_22446 stream (Lnet/minecraft/class_1923;I)Ljava/util/stream/Stream;
 		ARG 0 center
 		ARG 1 radius
+	METHOD method_30551 unpackLocalX (S)I
+		COMMENT Gets the local x-coordinate from the given packed local position.
```
Comment (haykam821) [inline mappings/net/minecraft/util/math/ChunkSectionPos.mapping:103]:
```suggestion
		COMMENT Gets the local X coordinate from the given packed local position.
```

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping [hunk 19]
```diff
@@ -73,6 +99,34 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
...
+		COMMENT @see #packLocal
+		ARG 0 packedLocalPos
+	METHOD method_30552 unpackLocalY (S)I
+		COMMENT Gets the local y-coordinate from the given packed local position.
```
Comment (haykam821) [inline mappings/net/minecraft/util/math/ChunkSectionPos.mapping:107]:
```suggestion
		COMMENT Gets the local Y coordinate from the given packed local position.
```

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping [hunk 20]
```diff
@@ -73,6 +99,34 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
...
+		COMMENT @see #packLocal
+		ARG 0 packedLocalPos
+	METHOD method_30553 unpackLocalZ (S)I
+		COMMENT Gets the local z-coordinate from the given packed local position.
```
Comment (haykam821) [inline mappings/net/minecraft/util/math/ChunkSectionPos.mapping:111]:
```suggestion
		COMMENT Gets the local Z coordinate from the given packed local position.
```

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping [hunk 21]
```diff
@@ -73,6 +99,34 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
...
+		COMMENT @see #packLocal
+		ARG 0 packedLocalPos
+	METHOD method_30554 unpackBlockX (S)I
+		COMMENT Gets the world x-coordinate of the given local position within this chunk section.
```
Comment (haykam821) [inline mappings/net/minecraft/util/math/ChunkSectionPos.mapping:115]:
```suggestion
		COMMENT Gets the world X coordinate of the given local position within this chunk section.
```

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping [hunk 22]
```diff
@@ -73,6 +99,34 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
...
+		COMMENT @see #packLocal
+		ARG 1 packedLocalPos
+	METHOD method_30555 unpackBlockY (S)I
+		COMMENT Gets the world y-coordinate of the given local position within this chunk section.
```
Comment (haykam821) [inline mappings/net/minecraft/util/math/ChunkSectionPos.mapping:119]:
```suggestion
		COMMENT Gets the world Y coordinate of the given local position within this chunk section.
```

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping [hunk 23]
```diff
@@ -73,6 +99,34 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
...
+		COMMENT @see #packLocal
+		ARG 1 packedLocalPos
+	METHOD method_30556 unpackBlockZ (S)I
+		COMMENT Gets the world z-coordinate of the given local position within this chunk section.
```
Comment (haykam821) [inline mappings/net/minecraft/util/math/ChunkSectionPos.mapping:123]:
```suggestion
		COMMENT Gets the world Z coordinate of the given local position within this chunk section.
```

**Review by haykam821 — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping (hunk 20 above)
Comment (shartte) [reply mappings/net/minecraft/util/math/ChunkSectionPos.mapping:111]: Is this actually correct? I couldn't find any resource that spells x-,y-,z-coordinate without the hyphen.

**Comment by shartte** [comment]
I've unified uses of x-coordinate, y-coordinate and z-coordinate and changed it to "retainLighting" over "skipLightingUpdates".
Also changed some uses of "chunk y-section" to just use "vertical chunk section" which is clearer.

**Review by haykam821 — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]

**Review by jaskarth — APPROVED** [review]

**Comment by modmuss50** [comment]
conflicts and then this is good to go. thanks.

**Review by liach — APPROVED** [review]

**Comment by shartte** [comment]
Okay, all fixed (of the conflicting methods, I only kept my name for one (setCompletedChunk rather than finalizeChunk)

**Review by Gegy — APPROVED** [review]

**Pull request merged**